packages/backend-common/src/paths.ts

```diff
@@ -18,6 +18,12 @@ import { isChildPath } from '@backstage/cli-common';
 import { NotAllowedError } from '@backstage/errors';
 import { resolve as resolvePath } from 'path';
 
+/** @internal */
+export const packagePathMocks = new Map<
```

Comment:
Heh, it's SO tempting to just do `resolvePackagePath.mock = overridePackagePathResolution`, is it possible to make the testUtils thing behave in such a way that you can import it in your setupTests and then you'll get that extra feature (added as a module in the type system too? 🤔 I DON'T KNOW THESE THINGS) injected.

It's just not super clear from the naming what the impact is from calling `overridePackagePathResolution` since the name kinda is rewritten from the one it affects. Not blocking on this at all tho

Reply:
Initial name was `createPackagePathMock`, then `installPackagePathMock`, then `mockPackagePathResolution`, now `overridePackagePathResolution`. Can go with either of those tbh 😅, keeping in mind the pattern that you often just call it once at the top and ignore the return value. I agree having `resolvePackagePath` involved would be optimal, but it feels a bit too icky to have it present in the main public API tbh.

Reply:
Let's keep considering a different API here, but gonna merge as is to move the overall work ahead